tests/test_client.py

```diff
@@ -1,11 +1,96 @@
+import multiprocessing
 import os
+import socket
+import time
+from collections.abc import Generator
 from pathlib import Path
 
 import pytest
+import uvicorn
 from langchain_core.messages import AIMessage
 from langchain_core.tools import BaseTool
+from mcp.server.websocket import websocket_server
+from starlette.applications import Starlette
+from starlette.routing import WebSocketRoute
 
 from langchain_mcp_adapters.client import MultiServerMCPClient
+from tests.servers.time_server import mcp as time_mcp
+
+
+@pytest.fixture
+def server_port() -> int:
+    with socket.socket() as s:
+        s.bind(("127.0.0.1", 0))
+        return s.getsockname()[1]
+    return -1
+
+
+@pytest.fixture
+def server_url(server_port: int) -> str:
+    return f"ws://127.0.0.1:{server_port}"
+
+
+def make_server_app() -> Starlette:
+    server = time_mcp._mcp_server
+
+    async def handle_ws(websocket):
+        async with websocket_server(websocket.scope, websocket.receive, websocket.send) as streams:
+            await server.run(streams[0], streams[1], server.create_initialization_options())
+
+    app = Starlette(
+        routes=[
+            WebSocketRoute("/ws", endpoint=handle_ws),
+        ]
+    )
+
+    return app
+
+
+def run_server(server_port: int) -> None:
+    app = make_server_app()
+    server = uvicorn.Server(
+        config=uvicorn.Config(app=app, host="127.0.0.1", port=server_port, log_level="error")
+    )
+    print(f"starting server on {server_port}")
+    server.run()
+
+    # Give server time to start
+    while not server.started:
+        print("waiting for server to start")
+        time.sleep(0.5)
+
+
+@pytest.fixture()
+def server(server_port: int) -> Generator[None, None, None]:
+    proc = multiprocessing.Process(
+        target=run_server, kwargs={"server_port": server_port}, daemon=True
+    )
+    print("starting process")
+    proc.start()
+
+    # Wait for server to be running
+    max_attempts = 20
+    attempt = 0
+    print("waiting for server to start")
+    while attempt < max_attempts:
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.connect(("127.0.0.1", server_port))
+                break
+        except ConnectionRefusedError:
+            time.sleep(0.1)
+            attempt += 1
+    else:
+        raise RuntimeError(f"Server failed to start after {max_attempts} attempts")
+
+    yield
+
+    print("killing server")
+    # Signal the server to stop
+    proc.kill()
+    proc.join(timeout=2)
+    if proc.is_alive():
+        print("server process failed to terminate")
 
 
 @pytest.mark.asyncio
```

Comment:
could we test it inside this unittest as well? and call the tool to make sure it's working?